### DISCUSSION TITLE
2020/02/how-to-deploy-github-pages-on-a-schedule-to-publish-future-posts/

### DISCUSSION THREAD

**Discussion by SeanKilleen** [SeanKilleen/seankilleen.github.io#640] [Specific Post Comments]
https://seankilleen.com/2020/02/how-to-deploy-github-pages-on-a-schedule-to-publish-future-posts/

**Comment by SeanKilleen** [comment]
## Historical Comments

2020-05-26, Disqus user [Pat Schloss](https://disqus.com/by/pschloss/):

> Thanks for these great instructions! In your post you put the workflow at `/github/workflows/scheduled-posts.yml` but in your repo, it's at `/.github/workflows/scheduled-posts.yml`. I assume the hidden directory is right?

2020-05-26, Me (responding to Pat):

> Hi Pat! I'm glad you found the article useful.
> 
> You are correct that the ".github" directory is the right one. I'll get the post updated. Thanks for catching it!

2020-06-24, Disqus User [Philippe Bourgau](https://disqus.com/by/philippebourgau/):

> Hi Sean. What I like to your solution is that it is very lightweight compared to the ton of github page deploy actions in the marketplace.
> I've just configured my blog to use your script. We are using the same theme by the way!
> Thanks again

2021-03-25, Disqus User [Jessica](https://disqus.com/by/jessicaliang721/):

> Hi Sean, where would you put the scheduled posts? Normally you'd have them sitting in a _drafts folder of some sort. Do you now have them under _posts, with a future date in the file name?

2021-03-25, Me (responding to Jessica):

> Hi Jessica! Yes, in this case they're not drafts but rather future-dated posts. The thing that takes them live is a build running after the date they're scheduled to be published, so by scheduling the build I can schedule taking them live. Let me know if that answers your question .Thanks!

**Comment by whatisityouseek** [comment]
This is really helpful. I'm running into a maddening problem. I've set future:false in my config.yml and when run locally it does omit future dated posts. However, on github pages, it doesn't seem to ignore future dated posts. One thing I turned up was that maybe the github pages gem is overriding this flag? Have you run into anything like that?

**Reply by SeanKilleen** [reply]
Interesting. I'll need to revisit my setup to answer this definitively, but you can see my current config and my GitHub actions that I use to publish the site [here at the GitHub repo](https://github.com/SeanKilleen/seankilleen.github.io). 

As far as I recall, the config setting for false and the future dated posts were all I needed. I had to create the scheduled action specifically to ensure pages would be published once the date/time has passed.

I'll double check this once I'm back at a PC.

**Comment by whatisityouseek** [comment]
Appreciate it! I hadn't looked at it in a month since I thought I had it all working and now I'm questioning my sanity. I remember that flag working perfectly which was why I needed the github action in the first place! Going to try rolling back some commits and see if it magically starts working again and then can debug.

**Comment by rupeshtiwari** [comment]
not sure why future: false is not working in my blog
Any comment advise Sean? It is showing all of my future blogs. 
https://github.com/rupeshtiwari/rupeshtiwari.github.io/actions 
https://www.rupeshtiwari.com/posts/

**Comment by FleXoft** [comment]
This is exactly what I have experienced. I opened an issue ticket at github without solution yet. I will try again.
